src/DurableTask.Netherite/StorageLayer/Faster/StoreWorker.cs

```diff
@@ -281,6 +282,42 @@ internal enum CheckpointTrigger
             Idle
         }
 
+        void LogCheckpointStats()
+        {
+            long inputQueuePositionLag = this.getInputQueuePositionLag();
+            this.CheckpointDue(out CheckpointTrigger trigger, out long? compactUntil);
+            var reportNullableTaskStatus = (Task? t) =>
+            {
+                if (t == null)
+                {
+                    return "is null";
+                }
+                else
+                {
+                    return t.IsCompleted ? "is completed" : "is not completed";
+                }
+            };
+
+            var checkpointDueLog = $"Checkpoint statistics: current checkpoint trigger={trigger}, " +
+                $"LastCheckpointedCommitLogPosition={this.lastCheckpointedCommitLogPosition}, " +
+                $"MaxNumberBytesBetweenCheckpoints={this.partition.Settings.MaxNumberBytesBetweenCheckpoints}, " +
+                $"CommitLogPosition={this.CommitLogPosition}, " +
+                $"NumberEventsSinceLastCheckpoint={this.numberEventsSinceLastCheckpoint}, " +
+                $"MaxNumberEventsBetweenCheckpoints={this.partition.Settings.MaxNumberEventsBetweenCheckpoints}, " +
+                $"InputQueuePositionLag={inputQueuePositionLag}," +
+                $"TimeOfNextIdleCheckpoint={this.timeOfNextIdleCheckpoint}, " +
+                $"TimeOfFirstRefusedCheckpoint={this.timeOfFirstRefusedCheckpoint}, " +
+                $"PendingCompaction status={reportNullableTaskStatus(this.pendingCompaction)}, " +
+                $"PendingIndexCheckpoint status={reportNullableTaskStatus(this.pendingIndexCheckpoint)}, " +
+                $"PendingStoreCheckpoint status={reportNullableTaskStatus(this.pendingStoreCheckpoint)}";
+            this.traceHelper.FasterProgress(checkpointDueLog);
+        }
+
+        long getInputQueuePositionLag()
```

Comment:
nit: method names in C# should always start with a capital letter.

Reply:
Ah yes, I sometimes misremember and assume that only applies to public methods, but I guess it also applies to private ones. Applied this feedback.